src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -186,6 +191,8 @@ private JoinableTask ExecuteWithinLockAsync(Func<Task> task)
             AggregateWorkspaceProjectContext previousContextToDispose = null;
             return await ExecuteWithinLockAsync(async () =>
             {
+                await _commonServices.ThreadingService.SwitchToUIThread();
```

Comment:
No, this is a common pattern in CPS code base. You create a joinable task, which ExecuteWithinLockAsync above does, and then switch to the UI thread for any UI operations. Without this change, GetConfigurationGeneralPropertiesAsync was throwing an exception on the original failing test.